sycl/include/sycl/exception.hpp

```diff
@@ -127,8 +127,16 @@ class __SYCL_EXPORT exception : public virtual std::exception {
   // base constructor for all SYCL 2020 constructors
   // exception(context *ctxPtr, std::error_code ec, const std::string
   // &what_arg);
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  exception(std::error_code ec, std::shared_ptr<context> SharedPtrCtx,
+            const std::string &what_arg)
+      : exception(ec, SharedPtrCtx, what_arg.c_str()) {}
+  exception(std::error_code EC, std::shared_ptr<context> SharedPtrCtx,
+            const char *WhatArg);
+#else
   exception(std::error_code ec, std::shared_ptr<context> SharedPtrCtx,
             const std::string &what_arg);
+#endif
```

Comment:
No, this is another issue in dual ABI usage case.
The exception object (which is construected using std::string) will need to cross the ABI boundary and the user program should handle the std::string created with a different ABI.
So, passing the native pointer will solve the issue. 
Whatever ABI the user program has, it will construct its ABI-compatible std::string.

Reply:
Great. Thanks for the explanation!